aiida/repository/backend/abstract.py

```diff
@@ -119,6 +119,23 @@ def list_objects(self) -> Iterable[str]:
         :return: An iterable for all the available object keys.
         """
 
+    def get_info(self, statistics=False, **kwargs) -> dict:  # pylint: disable=unused-argument,no-self-use
```

Comment:
Well, that is a good question.

I have not yet found a fully satisfactory criteria for when a method should be fully abstract or implement a default behavior. Right now I am more inclined to implement a default whenever possible, since it simplifies testing and prototyping. And technically, the maintain operation doesn't need to do anything (for example, if we implemented here our previous file repository, there is no maintenance to be done). This is opposed as something like the `put`/`get` methods, which actually have requirements that can't be given a default implementation.

I'm open to hearing different criterias though.

Reply:
> since it simplifies testing and prototyping.

That is only an argument in very rare cases, and here I definitely don't think it applies. Changing design or implementation just because it makes testing easier since you do not have to mock the methods for mock implementations, is not really a good approach I would say. Since maintain is expected to do something, I actually think this should be abstract, or at the very least the base class raises `NotImplementedError`. This prevents that you forget to implement it and an implementation actually doesn't do anything by mistake.

Reply:
> Changing design or implementation just because it makes testing easier since you do not have to mock the methods for mock implementations, is not really a good approach I would say.

Well, I agree, but this is not what I was saying:

> I have not yet found a fully satisfactory criteria for when a method should be fully abstract or implement a default behavior.

I am not using the argument to change an otherwise established convenient design. I am saying these two options seem otherwise equivalent to me in all other practical aspects, so that was the only distinction I found to decide on one vs the other.

If it was clear to me that the `maintain` method had anything that would be expected of it, I would have indeed chosen to do this an abstract method. But although it may be true that you typically call it because you expect it to do "something" (although this is also true of a lot of functions who can have "empty" results), since that can be extremely varied and different from backend to backend, I think the most robust thing is that clients of the feature operate under the assumption that it could actually do nothing (in which case setting that as the default is much more natural).

Anyways, having it raise a `NotImplementedError` is ok for me. It would mean clients have to add more safeguards when using it, making their code a bit clunkier, but they also get a more explicit treatment of the doing nothing case. I can't really say a priori if this will be a practically convenient tradeoff but we can try it out.